packages/firestore/test/integration/remote/stream.test.ts

```diff
@@ -173,68 +161,40 @@ class MockCredentialsProvider extends EmptyCredentialsProvider {
 }
 
 describe('Write Stream', () => {
-  let streamListener: StreamStatusListener;
-
-  beforeEach(() => {
-    streamListener = new StreamStatusListener();
-  });
-
-  afterEach(() => {
-    streamListener.verifyNoPendingCallbacks();
-  });
-
   /**
    * Verifies that the write stream issues an onClose callback after a call to
    * stop().
    */
   it('can be stopped before handshake', () => {
-    let writeStream: PersistentWriteStream;
-
-    return withTestDatastore(ds => {
-      writeStream = ds.newPersistentWriteStream(streamListener);
-      writeStream.start();
-      return streamListener.awaitCallback('open');
-    }).then(async () => {
+    return withTestWriteStream(async (writeStream, streamListener) => {
```

Comment:
Despite the other comment on thinking the newPersistentStream methods should stay in `Datastore`, this change is a good one (though `withTestWriteStream` should call `newPersistentWriteStream(datastore)`.

Reply:
Done